clients/agentic-sandbox-client-python/agentic_sandbox/sandbox.py

```diff
@@ -0,0 +1,188 @@
+CLAIM_API_GROUP = "extensions.agents.x-k8s.io"
+CLAIM_API_VERSION = "v1alpha1"
+CLAIM_PLURAL_NAME = "sandboxclaims"
+
+# Constants for Sandbox
+SANDBOX_API_GROUP = "agents.x-k8s.io"
+SANDBOX_API_VERSION = "v1alpha1"
+SANDBOX_PLURAL_NAME = "sandboxes"
+
+@dataclass
+class ExecutionResult:
+    """A structured object for holding the result of a command execution."""
+    stdout: str
+    stderr: str
+    exit_code: int
+
+class Sandbox:
+    """
+    The main client for creating and interacting with a stateful Sandbox.
+    This class is a context manager, designed to be used with a `with` statement.
+    """
+    def __init__(self, template_name: str, namespace: str = "default"):
+        self.template_name = template_name
+        self.namespace = namespace
+        self.claim_name: str | None = None
+        self.sandbox_name: str | None = None
+        self.port_forward_process: subprocess.Popen | None = None
+        self.server_port: int = 8888  # Assuming a default port
+
+        try:
+            config.load_incluster_config()
+        except config.ConfigException:
+            config.load_kube_config()
+
+        self.custom_objects_api = client.CustomObjectsApi()
+
+    def is_ready(self) -> bool:
+        """Returns True if the sandbox is created and ready for communication."""
+        return self.port_forward_process is not None
+
+    def __enter__(self) -> 'Sandbox':
+        """Creates the SandboxClaim resource and waits for the Sandbox to become ready."""
+        self.claim_name = f"sandbox-claim-{os.urandom(4).hex()}"
+        manifest = {
+            "apiVersion": f"{CLAIM_API_GROUP}/{CLAIM_API_VERSION}",
+            "kind": "SandboxClaim",
+            "metadata": {"name": self.claim_name},
+            "spec": {"sandboxTemplateRef": {"name": self.template_name}}
+        }
+
+        print(f"Creating SandboxClaim: {self.claim_name}...")
+        self.custom_objects_api.create_namespaced_custom_object(
+            group=CLAIM_API_GROUP,
+            version=CLAIM_API_VERSION,
+            namespace=self.namespace,
+            plural=CLAIM_PLURAL_NAME,
+            body=manifest
+        )
+
+        w = watch.Watch()
+        print("Watching for Sandbox to become ready...")
+        for event in w.stream(
+            func=self.custom_objects_api.list_namespaced_custom_object,
+            namespace=self.namespace,
+            group=SANDBOX_API_GROUP,
+            version=SANDBOX_API_VERSION,
+            plural=SANDBOX_PLURAL_NAME,
+            field_selector=f"metadata.name={self.claim_name}",
+            timeout_seconds=180
+        ):
+            sandbox_object = event['object']
+            status = sandbox_object.get('status', {})
+            conditions = status.get('conditions', [])
+            is_ready = False
+            for cond in conditions:
+                if cond.get('type') == 'Ready' and cond.get('status') == 'True':
+                    is_ready = True
+                    break
```

Comment:
The `timeout_seconds` for watching the sandbox is hardcoded to `180`. This should be a parameter to the `__enter__` method or the `__init__` method to allow users to configure the timeout based on their environment.